src/Core/AdminConsole/OrganizationFeatures/OrganizationUsers/DeleteManagedOrganizationUserAccountCommand.cs

```diff
@@ -50,140 +50,188 @@ public DeleteManagedOrganizationUserAccountCommand(
-        var orgUsersToDelete = orgUsers.Where(orgUser => orgUserResultsToDelete.Any(result => orgUser.Id == result.OrganizationUserId));
-        var usersToDelete = users.Where(user => orgUsersToDelete.Any(orgUser => orgUser.UserId == user.Id));
+        await HandleUserDeletionsAsync(userDeletionResults);
 
-        if (usersToDelete.Any())
-        {
-            await DeleteManyAsync(usersToDelete);
-        }
+        await LogDeletedOrganizationUsersAsync(userDeletionResults);
+
+        return userDeletionResults
+            .Select(i => (i.OrganizationUserId, i.ErrorMessage))
+            .ToList();
+    }
+
+    private async Task<IEnumerable<User>> GetUsersAsync(ICollection<OrganizationUser> orgUsers)
+    {
+        var userIds = orgUsers
+         .Where(orgUser => orgUser.UserId.HasValue)
+         .Select(orgUser => orgUser.UserId!.Value)
+         .ToList();
 
-        await LogDeletedOrganizationUsersAsync(orgUsers, results);
+        var users = await _userRepository.GetManyAsync(userIds);
 
-        return results;
+        return users;
     }
 
-    private async Task ValidateDeleteUserAsync(Guid organizationId, OrganizationUser orgUser, Guid? deletingUserId, IDictionary<Guid, bool> managementStatus, bool hasOtherConfirmedOwners)
+    private async Task ValidateDeleteUserAsync(Guid organizationId, OrganizationUser orgUser, User user, Guid? deletingUserId, IDictionary<Guid, bool> managementStatus, bool hasOtherConfirmedOwners)
+    {
+        EnsureUserStatusIsNotInvited(orgUser);
+        PreventSelfDeletion(orgUser, deletingUserId);
+        EnsureUserIsManagedByOrganization(orgUser, managementStatus);
+        PreventOrganizationSoleOwnerDeletion(orgUser, hasOtherConfirmedOwners);
+
+        await EnsureOnlyOwnersCanDeleteOwnersAsync(organizationId, orgUser, deletingUserId);
+        await EnsureUserIsNotSoleOrganizationOwnerAsync(user);
+        await EnsureUserIsNotSoleProviderOwnerAsync(user);
+    }
+    private static void EnsureUserStatusIsNotInvited(OrganizationUser orgUser)
     {
         if (!orgUser.UserId.HasValue || orgUser.Status == OrganizationUserStatusType.Invited)
         {
             throw new BadRequestException("You cannot delete a member with Invited status.");
         }
-
-        if (deletingUserId.HasValue && orgUser.UserId.Value == deletingUserId.Value)
+    }
+    private static void PreventSelfDeletion(OrganizationUser orgUser, Guid? deletingUserId)
+    {
+        if (!(orgUser.UserId.HasValue && deletingUserId.HasValue))
+        {
+            return;
+        }
+        if (orgUser.UserId.Value == deletingUserId.Value)
         {
             throw new BadRequestException("You cannot delete yourself.");
         }
+    }
 
-        if (orgUser.Type == OrganizationUserType.Owner)
+    private async Task EnsureOnlyOwnersCanDeleteOwnersAsync(Guid organizationId, OrganizationUser orgUser, Guid? deletingUserId)
```

Comment:
Maybe a nit? I think some of these validation methods could be a bit more concise, e.g.

Instead of:

```cs
        if (orgUser.Type != OrganizationUserType.Owner)
        {
            return;
        }

        if (deletingUserId.HasValue && !await _currentContext.OrganizationOwner(organizationId))
        {
            throw new BadRequestException("Only owners can delete other owners.");
        }
```

It could be:

```cs
        if (orgUser.Type == OrganizationUserType.Owner && 
          deletingUserId.HasValue &&
          !await _currentContext.OrganizationOwner(organizationId))
        {
            throw new BadRequestException("Only owners can delete other owners.");
        }
```

Reply:
```c#
        if (orgUser.Type != OrganizationUserType.Owner)
        {
            return;
        }

        if (deletingUserId.HasValue && !await _currentContext.OrganizationOwner(organizationId))
        {
            throw new BadRequestException("Only owners can delete other owners.");
        }
```

I like this option better because it reduces another variable I have to track, so by the time it gets to this point, I can focus on two variables instead of three. 

With that said, it's not something I feel strongly about. I'm fine with either option. Let me know if you want me to update it.

Reply:
I understand that argument - let's leave it how it is.